verify-sna/functions/check-verify.js

```diff
@@ -1,79 +1,243 @@
 
-    // TODO: Check that country code and phone number are present and correct
-
-    /**
-     * const check = await client.verify
-     * .services(service)
-     * .verificationChecks
-     * .create({to: countryCode + phoneNumber});
-     */
-
-    const check = {
-      status: 'approved',
-    };
+    const check = await client.verify.v2
+      .services(service)
+      .verificationChecks.create({ to: `${countryCode}${phoneNumber}` });
 
     if (check.status === 'approved') {
       response.setStatusCode(200);
       response.setBody({
-        message: 'Phone number verified successfully',
+        status: 'verified',
+        message: 'SNA verification successful, phone number verified',
       });
+    } else {
+      if (
+        check.snaAttemptsErrorCodes[check.snaAttemptsErrorCodes.length - 1]
+          .code === 60519 ||
+        check.snaAttemptsErrorCodes[check.snaAttemptsErrorCodes.length - 1]
+          .code > 60520
+      ) {
+        response.setStatusCode(200);
+        response.setBody({
+          status: 'pending',
+          message: 'Pending EVURL processing',
+        });
+      } else {
+        response.setStatusCode(200);
+        response.setBody({
+          status: 'not-verified',
+          message: 'SNA verification unsuccessful, phone number not verified',
+        });
+      }
+    }
 
-      // Send verification check details to rabbitMQ
-      amqp.connect(rabbitmqUrl, function (error0, connection) {
-        if (error0) {
-          throw error0;
-        }
-        connection.createChannel(function (error1, channel) {
-          if (error1) {
-            throw error1;
+    if (response.body.status !== 'pending') {
+      // Connecting to database and running queries
+      const dbName = 'verifications_db.db';
+      const db = new sqlite3.Database(
+        path.join(os.tmpdir(), dbName),
+        sqlite3.OPEN_READWRITE,
+        (err) => {
+          if (err && err.code === 'SQLITE_CANTOPEN') {
+            // Create database
+            const newdb = new sqlite3.Database(
+              path.join(os.tmpdir(), dbName),
+              (err) => {
+                if (err) {
+                  const statusCode = err.status || 400;
+                  response.setStatusCode(statusCode);
+                  response.setBody({
+                    message: err.message,
+                  });
+                  return callback(null, response);
+                }
+                // Table(s) creation
+                newdb.exec(
+                  `
+                        CREATE TABLE verifications (
+                            phone_number VARCHAR(30) NOT NULL,
+                            sna_url VARCHAR(500) NOT NULL,
+                            status VARCHAR(10) NOT NULL,
+                            verification_start_datetime DATETIME,
+                            verification_check_datetime DATETIME,
+                            PRIMARY KEY (phone_number, sna_url)
+                        );
```

Comment:
Sir, all this logic for creating the database when it does not exist gets duplicated. Can't we abstract this into a helper module that we can call from start, check and get-verifications ? Actually, we can abstract all the database logic in that file, clean up a bit the endpoints and make the code a bit more readable/maintainable. Thoughts?

Reply:
Yes sir that is a great idea, I will do that right away